### ISSUE TITLE
add_on_binder_for_namespace: place custom code after generated one

### ISSUE THREAD

**Issue by simleo** [RosettaCommons/binder#86]
Currently `add_on_binder_for_namespace` places the call to the function containing the custom manual bindings *before* binder-generated code. This means that the custom code cannot do several things, including:

* Bind a class that's a subclass of an automatically bound class, since that class is not yet known
* Set pybind11 default args to automatically bound types, since they haven't been registered yet

Thus, if there is no special reason to place the custom code before the generated one, it seems useful to do the opposite, so that the custom code can take advantage of what's already present in the generated one.

**Comment by lyskov** [comment]
@simleo i see your point. If i remember correctly there was a technical reason that made placing namespace binding on top, - i will see if it possible refactor this.

**Comment by lczech** [comment]
Hey @lyskov, running into the same issue here. I currently circumvent this by defining the bindings multiple times, once for each presence/absence of default arguments. Works, but a bit iffy.

Also, just in case that matters here: In my case, it would not be enough to just place the custom bindings at the end of the binding function that they are currently it, because the automatic type that the depend on is in a completely different bindings file. So the order of those files in the main.cpp also would need to change... Seems rather complicated to be honest. 

I've had a look at the sorting of bindings that Binder does before emitting the final generated code - that seems rather complicated, and I do not want to mess with that. But if you see a chance to easily fix this, that would be cool!

**Comment by lyskov** [comment]
> … Seems rather complicated to be honest. ...

@lczech yes, this is why this was originally tabled.  So far I have not found an obviously-right solution for this but if you have ideas please let me know.
